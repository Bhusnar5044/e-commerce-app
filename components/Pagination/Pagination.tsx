import usePagination from '@/hooks/usePagination';
import Link from 'next/link';
import { FC, memo } from 'react';
import { PaginationProps } from './types';

export const dotts = '...';

export const Pagination: FC<PaginationProps> = memo(
    ({ totalItems, currentPage, itemsPerPage = 20, renderPageLink }) => {
        const pages = usePagination(totalItems, currentPage, itemsPerPage);

        return (
            <div className="flex items-center justify-center my-8">
                {pages.map((pageNumber, i) =>
                    pageNumber === dotts ? (
                        <span key={i} className="px-4 py-2 rounded-full text-sm font-extralight text-black">
                            {pageNumber}
                        </span>
                    ) : (
                        <Link
                            key={i}
                            href={renderPageLink(pageNumber as number)}
                            className={`${
                                pageNumber === currentPage ? 'text-success-dark' : 'text-gray-500'
                            } px-4 py-2 mx-1 rounded-full text-sm font-semibold no-underline`}
                        >
                            {pageNumber}
                        </Link>
                    ),
                )}
            </div>
        );
    },
);

Pagination.displayName = 'Pagination';
