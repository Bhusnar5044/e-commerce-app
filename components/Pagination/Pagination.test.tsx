import { render } from '@test-utils';
import { Pagination } from './Pagination';

describe('Pagination', () => {
    it('should render properly', () => {
        const { container } = render(<Pagination totalItems={0} currentPage={0} renderPageLink={jest.fn()} />);
        expect(container).toMatchSnapshot();
    });
});
