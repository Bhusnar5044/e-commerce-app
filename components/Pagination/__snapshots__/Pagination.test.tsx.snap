// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render properly 1`] = `
<div>
  <div
    class="flex items-center justify-center my-8"
  />
</div>
`;
